host/CMakeLists.txt

```diff
@@ -259,18 +260,57 @@ if (OE_SGX)
 
   # OS specific as well.
   if (UNIX)
+    # Look for sgx.h from the kernel headers, which is required by vDSO support
+    # The header only exists after kernel version 5.11
+    execute_process(
+      COMMAND uname -r
+      OUTPUT_VARIABLE LINUX_KERNEL_VERSION
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+    message("-- The kernel version is ${LINUX_KERNEL_VERSION}")
+
+    set(LINUX_KERNEL_SGX_HEADER_PATH
+        "/lib/modules/${LINUX_KERNEL_VERSION}/build/arch/x86/include/uapi/asm")
+
+    message("-- Looking for sgx.h")
+
+    find_file(
+      LINUX_KERNEL_SGX_HEADER
+      NAMES "sgx.h"
+      PATHS ${LINUX_KERNEL_SGX_HEADER_PATH}
+      NO_DEFAULT_PATH)
+
+    if (LINUX_KERNEL_SGX_HEADER)
+      message("-- Looking for sgx.h - found")
+
+      file(COPY ${LINUX_KERNEL_SGX_HEADER}
```

Comment:
There are two alternatives:
1. Copy/mimic the vDSO data structure and function declaration into an OE header.
2. Cache the sgx.h

Both of the approaches are not ideal given that the file is under GPL. To avoid the license issues, I feel copying the header is simpler. From the release point of view, we should make sure that we build the SDK on the latest kernel. For the case of customized builds, the developers should manage the kernel version by themselves. What do you think?

Reply:
Fair points. @CyanDevs Do we build OE SDK packages on Ubuntu 20.04? I think ubuntu 18.04 doesn't have the neccesary kernel by default.

@mingweishih Can we copy the header from some repository?

Reply:
Yes we can copy from the linux github repository. However, I'm not sure which kernel version we should lock on though.

Reply:
The specific kernel version ought not matter. If it matters, then that means we cannot build OE in one kernel version and expect to work in another kernel version.

Reply:
Chagned to fetch the source from the github.